src/components/CopyTextToClipboard.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {TouchableOpacity} from 'react-native';
+import Text from './Text';
+import {Checkmark, Clipboard as ClipboardIcon} from './Icon/Expensicons';
+import Clipboard from '../libs/Clipboard';
+import Icon from './Icon';
+import styles from '../styles/styles';
+
+const propTypes = {
+    /** The text to display and copy to the clipboard */
+    text: PropTypes.string.isRequired,
+
+    /** Styles to apply to the text */
+    textStyles: PropTypes.arrayOf(PropTypes.object),
+};
+
+const defaultProps = {
+    textStyles: [],
+};
+
+class CopyTextToClipboard extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.copyToClipboard = this.copyToClipboard.bind(this);
+
+        this.state = {
+            showCheckmark: false,
+        };
+    }
+
+    componentWillUnmount() {
+        // Clear the interval when the component unmounts so that if the user navigates
+        // away quickly, then setState() won't try to update a component that's been unmounted
+        clearInterval(this.showCheckmarkInterval);
```

Comment:
Will this cause an error if `showCheckmarkInterval` hasn't been assigned before the clearInterval call?